src/main/java/com/gotocompany/depot/maxcompute/model/RecordWrapper.java

```diff
@@ -0,0 +1,18 @@
+package com.gotocompany.depot.maxcompute.model;
+
+import com.aliyun.odps.PartitionSpec;
+import com.aliyun.odps.data.Record;
+import com.gotocompany.depot.error.ErrorInfo;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+
+@AllArgsConstructor
+@Getter
+@Setter
+public class RecordWrapper {
+    private Record record;
+    private long index;
+    private ErrorInfo errorInfo;
+    private PartitionSpec partitionSpec;
+}
```

Comment:
The class is mutable with @Setter annotations which could lead to inconsistent state. The record wrapper should be immutable once created to prevent concurrent modification issues.

Reply:
sure, let me refactor this